data/transactions/logic/assembler.go

```diff
@@ -1541,73 +1563,77 @@ func (ops *OpStream) assemble(text string) error {
 	for scanner.Scan() {
 		ops.sourceLine++
 		line := scanner.Text()
-		line = strings.TrimSpace(line)
-		if len(line) == 0 {
-			ops.trace("%3d: 0 line\n", ops.sourceLine)
-			continue
-		}
-		if strings.HasPrefix(line, "//") {
-			ops.trace("%3d: // line\n", ops.sourceLine)
-			continue
-		}
-		if strings.HasPrefix(line, "#pragma") {
-			ops.trace("%3d: #pragma line\n", ops.sourceLine)
-			ops.pragma(line)
-			continue
-		}
 		fields := fieldsFromLine(line)
-		if len(fields) == 0 {
-			ops.trace("%3d: no fields\n", ops.sourceLine)
-			continue
-		}
-		// we're about to begin processing opcodes, so settle the Version
-		if ops.Version == assemblerNoVersion {
-			ops.Version = AssemblerDefaultVersion
-		}
-		if ops.versionedPseudoOps == nil {
-			ops.versionedPseudoOps = prepareVersionedPseudoTable(ops.Version)
-		}
-		opstring := fields[0]
-		if opstring[len(opstring)-1] == ':' {
-			ops.createLabel(opstring[:len(opstring)-1])
-			fields = fields[1:]
-			if len(fields) == 0 {
-				ops.trace("%3d: label only\n", ops.sourceLine)
+		for current, next := processFields(fields); len(current) > 0 || len(next) > 0; current, next = processFields(next) {
+			if len(current) == 0 {
 				continue
 			}
-			opstring = fields[0]
-		}
-		spec, expandedName, ok := getSpec(ops, opstring, fields[1:])
-		if ok {
-			ops.trace("%3d: %s\t", ops.sourceLine, opstring)
-			ops.recordSourceLine()
-			if spec.Modes == modeApp {
-				ops.HasStatefulOps = true
+			opstring := current[0]
+			if strings.HasPrefix(opstring, "//") {
+				ops.trace("%3d: comment\n", ops.sourceLine)
+				break
 			}
-			args, returns := spec.Arg.Types, spec.Return.Types
-			if spec.refine != nil {
-				nargs, nreturns := spec.refine(&ops.known, fields[1:])
-				if nargs != nil {
-					args = nargs
-				}
-				if nreturns != nil {
-					returns = nreturns
-				}
+			if opstring == "#pragma" {
+				ops.trace("%3d: #pragma line\n", ops.sourceLine)
+				// pragma get the rest of the tokens
```

Comment:
Why is this the case? This seems to allow "#pragma; version; 6"

Reply:
But current is the portion of the line that had no semis. They haven't really been stripped out, they weren't there.

Although you do make me realize that when we paste them back together, we need to insert a semi-colon if next is non-empty.

Reply:
Should be all good now